### PR TITLE
auto convert Python return value to Java value v2

### LINKED ISSUE #4562: Support numba @guvectorize vector functions in the query language
Numba's `guvectorize` annotation is extremely powerful.  It provides a few key features not found in `vectorize`, which is currently supported by DH.
1. Support for vector inputs
2. Support for vector outputs
3. Support for numpy inputs and operations

The following code illustrates what is possible.  The one notable unsupported feature from the numba side is fixed size outputs (https://github.com/numba/numba/issues/1668).

On the DH side, the query language needs to support outputs from the guvectorize functions.  Currently, all guvectorize output columns are `java.lang.Object` , instead of  being the proper scalar or array type.

Making this fix will allow high performance creation of native java arrays that can be seamlessly be used in operations such as grouping.  Also, the array support makes it easier to directly use inputs as numpy arrays.

```python

import numpy as np
from numba import vectorize, jit, guvectorize, int64

a = np.arange(5, dtype=np.int64)

@jit(int64(int64[:]),nopython=True)
def f(x):
    r = 0
    for xi in x:
        r += xi
    return r


print(f(a))

#vector input to scalar output function (m)->()
@guvectorize([(int64[:],int64[:])],"(m)->()",nopython=True)
def g(x, res):
    res[0] = 0
    for xi in x:
        res[0] += xi

print(g(a))

from deephaven import empty_table

t = empty_table(10).update(["X=i%3", "Y=i"]).group_by("X").update("Z=g(Y)")
m = t.meta_table

#vector and scalar input to vector ouput function
@guvectorize([(int64[:],int64,int64[:])],"(m),()->(m)",nopython=True)
def g2(x, y, res):
    for i in range(len(x)):
        res[i] = x[i] + y

print(g2(a,2))

t2 = empty_table(10).update(["X=i%3", "Y=i"]).group_by("X").update("Z=g2(Y,2)")
m2 = t2.meta_table

# NOTE: the following does not work according to this thread from 7 years ago:
# https://numba-users.continuum.narkive.com/7OAX8Suv/numba-guvectorize-with-fixed-size-output-array
# but the latest numpy Generalized Universal Function API does seem to support frozen dimensions
# https://numpy.org/doc/stable/reference/c-api/generalized-ufuncs.html#generalized-universal-function-api
# There is an old numba ticket about it
# https://github.com/numba/numba/issues/1668
# Possibly we could contribute a fix

# fails with: bad token in signature "2"

# #vector input to fixed-length vector ouput function
# @guvectorize([(int64[:],int64[:])],"(m)->(2)",nopython=True)
# def g3(x, res):
#     res[0] = min(x)
#     res[1] = max(x)

# print(g3(a))

# t3 = empty_table(10).update(["X=i%3", "Y=i"]).group_by("X").update("Z=g3(Y)")
# m3 = t3.meta_table

# ** Workaround **

dummy = np.array([0, 0], dtype=np.int64)

#vector input to fixed-length vector ouput function -- second arg is a dummy just to get a fixed size output
@guvectorize([(int64[:],int64[:],int64[:])],"(m),(n)->(n)",nopython=True)
def g4(x, dummy, res):
    res[0] = min(x)
    res[1] = max(x)

print(g4(a,dummy))

t4 = empty_table(10).update(["X=i%3", "Y=i"]).group_by("X").update("Z=g4(Y,dummy)")
m4 = t4.meta_table

# example using numpy

#vector input to fixed-length vector ouput function -- second arg is a dummy just to get a fixed size output
@guvectorize([(int64[:],int64[:],int64[:])],"(m),(n)->(n)",nopython=True)
def g5(x, dummy, res):
    res[0] = np.min(x)
    res[1] = np.max(x)

print(g5(a,dummy))

t5 = empty_table(10).update(["X=i%3", "Y=i"]).group_by("X").update("Z=g5(Y,dummy)")
m5 = t5.meta_table

# example using numpy

#vector input to fixed-length vector ouput function 
@guvectorize([(int64[:],int64[:])],"(m)->(m)",nopython=True)
def g6(x, res):
    res[:] = x+5

print(g6(a))

t6 = empty_table(10).update(["X=i%3", "Y=i"]).group_by("X").update("Z=g6(Y)")
m6 = t6.meta_table

```

### LINKED ISSUE #4569: Python type inference is not working
Type inference in python with type hints is not working:

```python
from deephaven import time_table

t = time_table("PT1s").update(["X=i%3","Y=i"]).group_by("X")

import numpy as np

def numpy_func(y) -> int:
    return int(np.sum(y))

t2 = t.update("Z=numpy_func(Y)")
```

This returns java.lang.Object for Z.

### LINKED ISSUE #4631: Autoconversion of python return values is buggy
#4579 is buggy.

This code runs fine:

```python
from deephaven import empty_table
import numpy as np
import numba as nb

t = empty_table(10).update(["A=i%2", "B=i"]).group_by("A")

# Testing https://github.com/deephaven/deephaven-core/issues/4557
def f4557_1(x, y) -> np.ndarray[np.int64]:
    # np.array is still needed as of v0.29
    return np.array(x) + y

# Testing https://github.com/deephaven/deephaven-core/issues/4562
@nb.guvectorize([(nb.int64[:],nb.int64,nb.int64[:])],"(m),()->(m)",nopython=True)
def f4562_1(x, y, res):
    res[:] = x + y

t2 = t.update([
    "X = f4557_1(B,3)",
    "Y = f4562_1(B,3)"
])

t3 = t2.ungroup()
```

But converting the example from #4557 yields a failed query
```python
from deephaven import empty_table
import numpy as np

def func() -> np.ndarray[np.int32]:
    return np.array([1,2,3], dtype=np.int32)

t = empty_table(10).update(["A = func()"])
m = t.meta_table
t2 = t.ungroup()
```
with the error
```
r-Scheduler-Serial-1 | .c.ConsoleServiceGrpcImpl | Error running script: java.lang.RuntimeException: Error in Python interpreter:
Type: <class 'deephaven.dherror.DHError'>
Value: table update operation failed. : RuntimeError: java.lang.RuntimeException: Error in Python interpreter:
Traceback (most recent call last):
  File "/opt/deephaven/venv/lib/python3.10/site-packages/deephaven/table.py", line 893, in update
    return Table(j_table=self.j_table.update(*formulas))
RuntimeError: java.lang.RuntimeException: Error in Python interpreter:
Type: <class 'ValueError'>
Value: cannot convert a Python 'numpy.ndarray' to a Java 'java.lang.Object'
Line: 487
Namespace: wrapper
File: /opt/deephaven/venv/lib/python3.10/site-packages/deephaven/table.py
Traceback (most recent call last):
  File "/opt/deephaven/venv/lib/python3.10/site-packages/deephaven/table.py", line 487, in wrapper

	at org.jpy.PyLib.callAndReturnValue(Native Method)
	at org.jpy.PyObject.call(PyObject.java:474)
	at io.deephaven.engine.table.impl.select.python.FormulaKernelPythonChunkedFunction.applyFormulaChunk(FormulaKernelPythonChunkedFunction.java:147)
	at io.deephaven.engine.table.impl.select.FormulaKernelTypedBase$ToTypedMethod.visit(FormulaKernelTypedBase.java:174)
	at io.deephaven.chunk.ObjectChunk.walk(ObjectChunk.java:130)
	at io.deephaven.engine.table.impl.select.FormulaKernelTypedBase.applyFormulaChunk(FormulaKernelTypedBase.java:42)
	at io.deephaven.engine.table.impl.select.formula.FormulaKernelAdapter.fillChunkHelper(FormulaKernelAdapter.java:299)
	at io.deephaven.engine.table.impl.select.formula.FormulaKernelAdapter.fillChunk(FormulaKernelAdapter.java:229)
	at io.deephaven.engine.table.impl.sources.ViewColumnSource.fillChunk(ViewColumnSource.java:219)
	at io.deephaven.engine.table.impl.select.analyzers.SelectColumnLayer.doApplyUpdate(SelectColumnLayer.java:412)
	at io.deephaven.engine.table.impl.select.analyzers.SelectColumnLayer.lambda$doSerialApplyUpdate$2(SelectColumnLayer.java:264)
	at io.deephaven.engine.util.systemicmarking.SystemicObjectTracker.executeSystemically(SystemicObjectTracker.java:56)
	at io.deephaven.engine.table.impl.select.analyzers.SelectColumnLayer.doSerialApplyUpdate(SelectColumnLayer.java:263)
	at io.deephaven.engine.table.impl.select.analyzers.SelectColumnLayer$1.lambda$onAllRequiredColumnsCompleted$1(SelectColumnLayer.java:212)
	at io.deephaven.engine.table.impl.util.ImmediateJobScheduler.submit(ImmediateJobScheduler.java:23)
	at io.deephaven.engine.table.impl.select.analyzers.SelectColumnLayer$1.onAllRequiredColumnsCompleted(SelectColumnLayer.java:210)
	at io.deephaven.engine.table.impl.select.analyzers.SelectAndViewAnalyzer$SelectLayerCompletionHandler.onLayerCompleted(SelectAndViewAnalyzer.java:589)
	at io.deephaven.engine.table.impl.select.analyzers.BaseLayer.applyUpdate(BaseLayer.java:76)
	at io.deephaven.engine.table.impl.select.analyzers.SelectColumnLayer.applyUpdate(SelectColumnLayer.java:151)
	at io.deephaven.engine.table.impl.QueryTable.lambda$selectOrUpdate$30(QueryTable.java:1501)
	at io.deephaven.engine.table.impl.perf.QueryPerformanceRecorder.withNugget(QueryPerformanceRecorder.java:519)
	at io.deephaven.engine.table.impl.QueryTable.lambda$selectOrUpdate$31(QueryTable.java:1451)
	at io.deephaven.engine.table.impl.QueryTable.memoizeResult(QueryTable.java:3457)
	at io.deephaven.engine.table.impl.QueryTable.selectOrUpdate(QueryTable.java:1450)
	at io.deephaven.engine.table.impl.QueryTable.update(QueryTable.java:1428)
	at io.deephaven.engine.table.impl.QueryTable.update(QueryTable.java:109)
	at io.deephaven.api.TableOperationsDefaults.update(TableOperationsDefaults.java:94)
	at org.jpy.PyLib.executeCode(Native Method)
	at org.jpy.PyObject.executeCode(PyObject.java:138)
	at io.deephaven.engine.util.PythonEvaluatorJpy.evalScript(PythonEvaluatorJpy.java:73)
	at io.deephaven.integrations.python.PythonDeephavenSession.lambda$evaluate$1(PythonDeephavenSession.java:194)
	at io.deephaven.util.locks.FunctionalLock.doLockedInterruptibly(FunctionalLock.java:51)
	at io.deephaven.integrations.python.PythonDeephavenSession.evaluate(PythonDeephavenSession.java:194)
	at io.deephaven.engine.util.AbstractScriptSession.lambda$evaluateScript$0(AbstractScriptSession.java:146)
	at io.deephaven.engine.context.ExecutionContext.lambda$apply$0(ExecutionContext.java:149)
	at io.deephaven.engine.context.ExecutionContext.apply(ExecutionContext.java:160)
	at io.deephaven.engine.context.ExecutionContext.apply(ExecutionContext.java:148)
	at io.deephaven.engine.util.AbstractScriptSession.evaluateScript(AbstractScriptSession.java:146)
	at io.deephaven.engine.util.DelegatingScriptSession.evaluateScript(DelegatingScriptSession.java:84)
	at io.deephaven.engine.util.ScriptSession.evaluateScript(ScriptSession.java:102)
	at io.deephaven.server.console.ConsoleServiceGrpcImpl.lambda$executeCommand$4(ConsoleServiceGrpcImpl.java:175)
	at io.deephaven.server.session.SessionState$ExportBuilder.lambda$submit$2(SessionState.java:1384)
	at io.deephaven.server.session.SessionState$ExportObject.doExport(SessionState.java:920)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at io.deephaven.server.runner.scheduler.SchedulerModule$ThreadFactory.lambda$newThread$0(SchedulerModule.java:78)
	at org.jpy.PyLib.callAndReturnObject(Native Method)
	at org.jpy.PyObject.call(PyObject.java:449)
	at io.deephaven.server.console.python.DebuggingInitializer.lambda$createInitializer$0(DebuggingInitializer.java:30)
	at java.base/java.lang.Thread.run(Thread.java:833)


Line: 895
Namespace: update
File: /opt/deephaven/venv/lib/python3.10/site-packages/deephaven/table.py
Traceback (most recent call last):
  File "<string>", line 7, in <module>
  File "/opt/deephaven/venv/lib/python3.10/site-packages/deephaven/table.py", line 895, in update

	at org.jpy.PyLib.executeCode(Native Method)
	at org.jpy.PyObject.executeCode(PyObject.java:138)
	at io.deephaven.engine.util.PythonEvaluatorJpy.evalScript(PythonEvaluatorJpy.java:73)
	at io.deephaven.integrations.python.PythonDeephavenSession.lambda$evaluate$1(PythonDeephavenSession.java:194)
	at io.deephaven.util.locks.FunctionalLock.doLockedInterruptibly(FunctionalLock.java:51)
	at io.deephaven.integrations.python.PythonDeephavenSession.evaluate(PythonDeephavenSession.java:194)
	at io.deephaven.engine.util.AbstractScriptSession.lambda$evaluateScript$0(AbstractScriptSession.java:146)
	at io.deephaven.engine.context.ExecutionContext.lambda$apply$0(ExecutionContext.java:149)
	at io.deephaven.engine.context.ExecutionContext.apply(ExecutionContext.java:160)
	at io.deephaven.engine.context.ExecutionContext.apply(ExecutionContext.java:148)
	at io.deephaven.engine.util.AbstractScriptSession.evaluateScript(AbstractScriptSession.java:146)
	at io.deephaven.engine.util.DelegatingScriptSession.evaluateScript(DelegatingScriptSession.java:84)
	at io.deephaven.engine.util.ScriptSession.evaluateScript(ScriptSession.java:102)
	at io.deephaven.server.console.ConsoleServiceGrpcImpl.lambda$executeCommand$4(ConsoleServiceGrpcImpl.java:175)
	at io.deephaven.server.session.SessionState$ExportBuilder.lambda$submit$2(SessionState.java:1384)
	at io.deephaven.server.session.SessionState$ExportObject.doExport(SessionState.java:920)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at io.deephaven.server.runner.scheduler.SchedulerModule$ThreadFactory.lambda$newThread$0(SchedulerModule.java:78)
	at org.jpy.PyLib.callAndReturnObject(Native Method)
	at org.jpy.PyObject.call(PyObject.java:449)
	at io.deephaven.server.console.python.DebuggingInitializer.lambda$createInitializer$0(DebuggingInitializer.java:30)
	at java.base/java.lang.Thread.run(Thread.java:833)

```

Additionally, when I add a dummy parameter, a different error is produced:
```python
from deephaven import empty_table
import numpy as np

def func(xx) -> np.ndarray[np.int32]:
    return np.array([1,2,3], dtype=np.int32)

t = empty_table(10).update(["A = func(3)"])
m = t.meta_table
t2 = t.ungroup()
```
with the error:
```
r-Scheduler-Serial-1 | .c.ConsoleServiceGrpcImpl | Error running script: java.lang.RuntimeException: Error in Python interpreter:
Type: <class 'deephaven.dherror.DHError'>
Value: table update operation failed. : RuntimeError: java.lang.RuntimeException: Error in Python interpreter:
Traceback (most recent call last):
  File "/opt/deephaven/venv/lib/python3.10/site-packages/deephaven/table.py", line 893, in update
    return Table(j_table=self.j_table.update(*formulas))
RuntimeError: java.lang.RuntimeException: Error in Python interpreter:
Type: <class 'TypeError'>
Value: only length-1 arrays can be converted to Python scalars
Line: 329
Namespace: _scalar
File: /opt/deephaven/venv/lib/python3.10/site-packages/deephaven/dtypes.py
Traceback (most recent call last):
  File "/opt/deephaven/venv/lib/python3.10/site-packages/deephaven/table.py", line 484, in wrapper
  File "/opt/deephaven/venv/lib/python3.10/site-packages/deephaven/dtypes.py", line 329, in _scalar

	at org.jpy.PyLib.callAndReturnValue(Native Method)
	at org.jpy.PyObject.call(PyObject.java:474)
	at io.deephaven.engine.table.impl.select.python.FormulaKernelPythonChunkedFunction.applyFormulaChunk(FormulaKernelPythonChunkedFunction.java:147)
	at io.deephaven.engine.table.impl.select.FormulaKernelTypedBase$ToTypedMethod.visit(FormulaKernelTypedBase.java:174)
	at io.deephaven.chunk.ObjectChunk.walk(ObjectChunk.java:130)
	at io.deephaven.engine.table.impl.select.FormulaKernelTypedBase.applyFormulaChunk(FormulaKernelTypedBase.java:42)
	at io.deephaven.engine.table.impl.select.formula.FormulaKernelAdapter.fillChunkHelper(FormulaKernelAdapter.java:299)
	at io.deephaven.engine.table.impl.select.formula.FormulaKernelAdapter.fillChunk(FormulaKernelAdapter.java:229)
	at io.deephaven.engine.table.impl.sources.ViewColumnSource.fillChunk(ViewColumnSource.java:219)
	at io.deephaven.engine.table.impl.select.analyzers.SelectColumnLayer.doApplyUpdate(SelectColumnLayer.java:412)
	at io.deephaven.engine.table.impl.select.analyzers.SelectColumnLayer.lambda$doSerialApplyUpdate$2(SelectColumnLayer.java:264)
	at io.deephaven.engine.util.systemicmarking.SystemicObjectTracker.executeSystemically(SystemicObjectTracker.java:56)
	at io.deephaven.engine.table.impl.select.analyzers.SelectColumnLayer.doSerialApplyUpdate(SelectColumnLayer.java:263)
	at io.deephaven.engine.table.impl.select.analyzers.SelectColumnLayer$1.lambda$onAllRequiredColumnsCompleted$1(SelectColumnLayer.java:212)
	at io.deephaven.engine.table.impl.util.ImmediateJobScheduler.submit(ImmediateJobScheduler.java:23)
	at io.deephaven.engine.table.impl.select.analyzers.SelectColumnLayer$1.onAllRequiredColumnsCompleted(SelectColumnLayer.java:210)
	at io.deephaven.engine.table.impl.select.analyzers.SelectAndViewAnalyzer$SelectLayerCompletionHandler.onLayerCompleted(SelectAndViewAnalyzer.java:589)
	at io.deephaven.engine.table.impl.select.analyzers.BaseLayer.applyUpdate(BaseLayer.java:76)
	at io.deephaven.engine.table.impl.select.analyzers.SelectColumnLayer.applyUpdate(SelectColumnLayer.java:151)
	at io.deephaven.engine.table.impl.QueryTable.lambda$selectOrUpdate$30(QueryTable.java:1501)
	at io.deephaven.engine.table.impl.perf.QueryPerformanceRecorder.withNugget(QueryPerformanceRecorder.java:519)
	at io.deephaven.engine.table.impl.QueryTable.lambda$selectOrUpdate$31(QueryTable.java:1451)
	at io.deephaven.engine.table.impl.QueryTable.memoizeResult(QueryTable.java:3457)
	at io.deephaven.engine.table.impl.QueryTable.selectOrUpdate(QueryTable.java:1450)
	at io.deephaven.engine.table.impl.QueryTable.update(QueryTable.java:1428)
	at io.deephaven.engine.table.impl.QueryTable.update(QueryTable.java:109)
	at io.deephaven.api.TableOperationsDefaults.update(TableOperationsDefaults.java:94)
	at org.jpy.PyLib.executeCode(Native Method)
	at org.jpy.PyObject.executeCode(PyObject.java:138)
	at io.deephaven.engine.util.PythonEvaluatorJpy.evalScript(PythonEvaluatorJpy.java:73)
	at io.deephaven.integrations.python.PythonDeephavenSession.lambda$evaluate$1(PythonDeephavenSession.java:194)
	at io.deephaven.util.locks.FunctionalLock.doLockedInterruptibly(FunctionalLock.java:51)
	at io.deephaven.integrations.python.PythonDeephavenSession.evaluate(PythonDeephavenSession.java:194)
	at io.deephaven.engine.util.AbstractScriptSession.lambda$evaluateScript$0(AbstractScriptSession.java:146)
	at io.deephaven.engine.context.ExecutionContext.lambda$apply$0(ExecutionContext.java:149)
	at io.deephaven.engine.context.ExecutionContext.apply(ExecutionContext.java:160)
	at io.deephaven.engine.context.ExecutionContext.apply(ExecutionContext.java:148)
	at io.deephaven.engine.util.AbstractScriptSession.evaluateScript(AbstractScriptSession.java:146)
	at io.deephaven.engine.util.DelegatingScriptSession.evaluateScript(DelegatingScriptSession.java:84)
	at io.deephaven.engine.util.ScriptSession.evaluateScript(ScriptSession.java:102)
	at io.deephaven.server.console.ConsoleServiceGrpcImpl.lambda$executeCommand$4(ConsoleServiceGrpcImpl.java:175)
	at io.deephaven.server.session.SessionState$ExportBuilder.lambda$submit$2(SessionState.java:1384)
	at io.deephaven.server.session.SessionState$ExportObject.doExport(SessionState.java:920)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at io.deephaven.server.runner.scheduler.SchedulerModule$ThreadFactory.lambda$newThread$0(SchedulerModule.java:78)
	at org.jpy.PyLib.callAndReturnObject(Native Method)
	at org.jpy.PyObject.call(PyObject.java:449)
	at io.deephaven.server.console.python.DebuggingInitializer.lambda$createInitializer$0(DebuggingInitializer.java:30)
	at java.base/java.lang.Thread.run(Thread.java:833)


Line: 895
Namespace: update
File: /opt/deephaven/venv/lib/python3.10/site-packages/deephaven/table.py
Traceback (most recent call last):
  File "<string>", line 7, in <module>
  File "/opt/deephaven/venv/lib/python3.10/site-packages/deephaven/table.py", line 895, in update

	at org.jpy.PyLib.executeCode(Native Method)
	at org.jpy.PyObject.executeCode(PyObject.java:138)
	at io.deephaven.engine.util.PythonEvaluatorJpy.evalScript(PythonEvaluatorJpy.java:73)
	at io.deephaven.integrations.python.PythonDeephavenSession.lambda$evaluate$1(PythonDeephavenSession.java:194)
	at io.deephaven.util.locks.FunctionalLock.doLockedInterruptibly(FunctionalLock.java:51)
	at io.deephaven.integrations.python.PythonDeephavenSession.evaluate(PythonDeephavenSession.java:194)
	at io.deephaven.engine.util.AbstractScriptSession.lambda$evaluateScript$0(AbstractScriptSession.java:146)
	at io.deephaven.engine.context.ExecutionContext.lambda$apply$0(ExecutionContext.java:149)
	at io.deephaven.engine.context.ExecutionContext.apply(ExecutionContext.java:160)
	at io.deephaven.engine.context.ExecutionContext.apply(ExecutionContext.java:148)
	at io.deephaven.engine.util.AbstractScriptSession.evaluateScript(AbstractScriptSession.java:146)
	at io.deephaven.engine.util.DelegatingScriptSession.evaluateScript(DelegatingScriptSession.java:84)
	at io.deephaven.engine.util.ScriptSession.evaluateScript(ScriptSession.java:102)
	at io.deephaven.server.console.ConsoleServiceGrpcImpl.lambda$executeCommand$4(ConsoleServiceGrpcImpl.java:175)
	at io.deephaven.server.session.SessionState$ExportBuilder.lambda$submit$2(SessionState.java:1384)
	at io.deephaven.server.session.SessionState$ExportObject.doExport(SessionState.java:920)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at io.deephaven.server.runner.scheduler.SchedulerModule$ThreadFactory.lambda$newThread$0(SchedulerModule.java:78)
	at org.jpy.PyLib.callAndReturnObject(Native Method)
	at org.jpy.PyObject.call(PyObject.java:449)
	at io.deephaven.server.console.python.DebuggingInitializer.lambda$createInitializer$0(DebuggingInitializer.java:30)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

### PR DESCRIPTION
**Pull request by jmao-denver**
Only support 1d array and scalars for now.

Fixes #4562 
Fixes #4569
Fixes #4631

### CODE REVIEW

File: engine/table/src/main/java/io/deephaven/engine/util/PyCallableWrapperJpyImpl.java [hunk 1]
```diff
@@ -40,6 +41,16 @@ public class PyCallableWrapperJpyImpl implements PyCallableWrapper {
         numpyType2JavaClass.put('O', Object.class);
     }
 
+    // TODO: add support for vectorizing functions that return arrays
```
Comment (chipkent) [inline engine/table/src/main/java/io/deephaven/engine/util/PyCallableWrapperJpyImpl.java:44]: ticket to increase the supported types?

**Review by chipkent — APPROVED** [review]

File: engine/table/src/main/java/io/deephaven/engine/table/impl/lang/QueryLanguageParser.java [hunk 2]
```diff
@@ -2688,6 +2693,7 @@ private void checkVectorizability(@NotNull final MethodCallExpr n,
             throw new PythonCallVectorizationFailure("Python function argument count mismatch: " + n + " "
                     + paramTypes.size() + " vs. " + expressions.length);
         }
+
```
Comment (rcaudy) [inline engine/table/src/main/java/io/deephaven/engine/table/impl/lang/QueryLanguageParser.java:2696]: Delete please

File: engine/table/src/main/java/io/deephaven/engine/util/PyCallableWrapperJpyImpl.java [hunk 3]
```diff
@@ -40,6 +41,16 @@ public class PyCallableWrapperJpyImpl implements PyCallableWrapper {
...
+    private static final Set<Class<?>> vectorizableReturnTypes = Set.of(int.class, long.class, short.class, float.class,
```
Comment (rcaudy) [inline engine/table/src/main/java/io/deephaven/engine/util/PyCallableWrapperJpyImpl.java:45]: No primitive chars?

Comment (rcaudy) [reply engine/table/src/main/java/io/deephaven/engine/util/PyCallableWrapperJpyImpl.java:45]: You verbally said we can't support them.

**Review by rcaudy — APPROVED** [review]

**Comment by deephaven-internal** [comment]
Labels indicate documentation is required. Issues for documentation have been opened:

How-to: https://github.com/deephaven/deephaven.io/issues/3309
Reference: https://github.com/deephaven/deephaven.io/issues/3308

**Pull request merged**